Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : Button
+	{
+		static Dictionary<string, List<WeakReference<RadioButton>>> _groupNameToElements;
+
+		public const string IsCheckedVisualState = "IsChecked";
+
+		public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create(
+			nameof(IsChecked), typeof(bool), typeof(RadioButton), false, propertyChanged: (b, o, n) => ((RadioButton)b).OnIsCheckedPropertyChanged((bool)n));
+
+		public static readonly BindableProperty GroupNameProperty = BindableProperty.Create(
+			nameof(GroupName), typeof(string), typeof(RadioButton), null, propertyChanged: (b, o, n) => ((RadioButton)b).OnGroupNamePropertyChanged((string)o, (string)n));
+
+		public static readonly BindableProperty ButtonSourceProperty = BindableProperty.Create(
+			nameof(ButtonSource), typeof(ImageSource), typeof(RadioButton), null);
+
+		public event EventHandler<CheckedChangedEventArgs> CheckedChanged;
+
+		public bool IsChecked
```

Comment:
`Checked` is probably a better name. We try to avoid property names containing multiple verbs. The `IsXxxx` construct is valid when Xxxx is an adjective. We can debate that in this case `Checked` is used as an adjective or not, I still think the name is overloaded

Reply:
this is not a big deal, I know we have `IsEnabled` already, and might want to reserve the `Checked` name for the event. so it's fine to leave it as is, and rename the event